File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int n, int[] slicer, int[] num_list) 
    {
        List<int> num = new List<int>();
        int a = slicer[0];
        int b = slicer[1];
        int c = slicer[2];
        
        switch (n) {
            case 1:
                for (int i=0; i<=b; i++) {
                    num.Add(num_list[i]);
                }
                break;
            case 2:
                for (int i=a; i<num_list.Length; i++) {
                    num.Add(num_list[i]);
                }
                break;
            case 3:
                for (int i=a; i<=b; i++) {
                    num.Add(num_list[i]);
                }
                break;
            case 4:
                    for (int i=a; i<=b; i = i+c) {
                    num.Add(num_list[i]);
                }
                break;
        }
        int[] answer = num.ToArray();
        return answer;
    }
}

File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, vector<int> slicer, vector<int> num_list) 
{
    vector<int> v;
    switch(n)
    {
        case 1:
            for(int i=0; i<=slicer[1]; i++)
            {
                v.push_back(num_list[i]);
            }
            break;
        case 2:
            for(int i=slicer[0]; i<num_list.size(); i++)
            {
                v.push_back(num_list[i]);
            }
            break;
        case 3:
            for(int i=slicer[0]; i<=slicer[1]; i++)
            {
                v.push_back(num_list[i]);
            }
            break;
        case 4:
            for(int i=slicer[0]; i<=slicer[1]; i += slicer[2])
            {
                v.push_back(num_list[i]);
            }
            break;
    }
    return v;
}